### PR TITLE
Additional debugging for WaitForOpenShiftNamespaceImageStreams

### PR DESCRIPTION
**Pull request by coreydaley**
Adding additional debugging information to diagnose a
 failure running WaitForOpenShiftNamespaceImageStreams

### CODE REVIEW

**Comment by coreydaley** [comment]
[testextended][extended:core(should fail resolving unresolvable valueFrom in sti build environment variable references)]

**Comment by coreydaley** [comment]
@bparees fyi

**Comment by coreydaley** [comment]
[test]

**Comment by coreydaley** [comment]
[testextended][extended:core(process valueFrom in build strategy environment variables)]

File: test/extended/util/framework.go [hunk 1]
```diff
@@ -50,15 +50,20 @@ const pvPrefix = "pv-"
...
+			fmt.Fprintf(g.GinkgoWriter, "%v: Checking %v \n", fnName, lang)
 			is, err := oc.Client().ImageStreams("openshift").Get(lang, metav1.GetOptions{})
 			if err != nil {
+				fmt.Fprintf(g.GinkgoWriter, "%v: IS Error: %#v \n", fnName, err)
 				return false
 			}
 			for tag := range is.Spec.Tags {
+				fmt.Fprintf(g.GinkgoWriter, "%v: Checking tag %v \n", fnName, tag)
 				if _, ok := is.Status.Tags[tag]; !ok {
+					fmt.Fprintf(g.GinkgoWriter, "%v: Tag Error: %#v \n", fnName, err)
```
Comment (bparees) [inline test/extended/util/framework.go]: you're not going to get any debugging output if ok is false.  you should print the value of ok explicitly.

**Comment by bparees** [comment]
i'd also update dumpimagestreams to dump the full yaml instead of what it does today.  that will be more useful if there's a failure to import an imagestream or something (hopefully the imagestream status will be informative about why it failed to import)

**Comment by coreydaley** [comment]
[testextended][extended:core(process valueFrom in build strategy environment variables)]

**Comment by coreydaley** [comment]
[testextended][extended:core(process valueFrom in build strategy environment variables)]

**Comment by coreydaley** [comment]
[testextended][extended:core(process valueFrom in build strategy environment variables)]

**Comment by coreydaley** [comment]
[testextended][extended:core(builds)]

**Comment by coreydaley** [comment]
Switching to running the whole builds suite as just running the valueFrom tests doesn't seem to be generating the failure that I need to debug this.

**Comment by coreydaley** [comment]
[testextended][extended:core(builds)]

**Comment by coreydaley** [comment]
[testextended][extended:core(builds)]

**Comment by coreydaley** [comment]
[testextended][extended:core(builds)]

**Comment by coreydaley** [comment]
@openshift/devex ptal

**Comment by coreydaley** [comment]
[test]

File: test/extended/util/framework.go [hunk 2]
```diff
@@ -110,17 +117,17 @@ func CheckOpenShiftNamespaceImageStreams(oc *CLI) {
...
 func DumpImageStreams(oc *CLI) {
-	out, err := oc.Run("get").Args("is", "-n", "openshift", "--config", KubeConfigPath()).Output()
+	out, err := oc.Run("get").Args("is", "-n", "openshift", "-o", "yaml", "--config", KubeConfigPath()).Output()
 	if err == nil {
 		fmt.Fprintf(g.GinkgoWriter, "\n  imagestreams in openshift namespace: \n%s\n", out)
 	} else {
-		fmt.Fprintf(g.GinkgoWriter, "\n  error on getting imagestreams in openshift namespace: %+v\n", err)
+		fmt.Fprintf(g.GinkgoWriter, "\n  error on getting imagestreams in openshift namespace: %+v\n%#v\n", err, out)
 	}
 	out, err = oc.Run("get").Args("is").Output()
```
Comment (bparees) [inline test/extended/util/framework.go]: @coreydaley make this a -o yaml too.

Comment (coreydaley) [inline test/extended/util/framework.go]: updated

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
[merge]

**Comment by bparees** [comment]
failure is known flake and won't conflict w/ anything coming. manual merge.

**Pull request merged**